homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,338 @@
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.util import slugify
+
+from .camera import generate_auth
+from .const import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_RTSP_TRANSPORT,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    DEFAULT_NAME,
+    DOMAIN,
+    FFMPEG_OPTION_MAP,
+    GET_IMAGE_TIMEOUT,
+    RTSP_TRANSPORTS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DATA = {
+    CONF_NAME: DEFAULT_NAME,
+    CONF_AUTHENTICATION: HTTP_BASIC_AUTHENTICATION,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE: False,
+    CONF_FRAMERATE: 2,
+    CONF_VERIFY_SSL: True,
+}
+
+SUPPORTED_IMAGE_TYPES = ["png", "jpeg", "svg+xml"]
+
+
+def build_schema(
+    user_input: dict[str, Any] | MappingProxyType[str, Any],
+    is_options_flow: bool = False,
+):
+    """Create schema for camera config setup."""
+    spec = {
+        vol.Optional(
+            CONF_STILL_IMAGE_URL,
+            description={"suggested_value": user_input.get(CONF_STILL_IMAGE_URL, "")},
+        ): str,
+        vol.Optional(
+            CONF_STREAM_SOURCE,
+            description={"suggested_value": user_input.get(CONF_STREAM_SOURCE, "")},
+        ): str,
+        vol.Optional(
+            CONF_RTSP_TRANSPORT,
+            description={"suggested_value": user_input.get(CONF_RTSP_TRANSPORT)},
+        ): vol.In(RTSP_TRANSPORTS),
+        vol.Optional(
+            CONF_AUTHENTICATION,
+            description={"suggested_value": user_input.get(CONF_AUTHENTICATION)},
+        ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+        vol.Optional(
+            CONF_USERNAME,
+            description={"suggested_value": user_input.get(CONF_USERNAME, "")},
+        ): str,
+        vol.Optional(
+            CONF_PASSWORD,
+            description={"suggested_value": user_input.get(CONF_PASSWORD, "")},
+        ): str,
+        vol.Required(
+            CONF_FRAMERATE,
+            description={"suggested_value": user_input.get(CONF_FRAMERATE, 2)},
+        ): int,
+        vol.Required(
+            CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL, True)
+        ): bool,
+    }
+    if is_options_flow:
+        spec[
+            vol.Required(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE, False),
+            )
+        ] = bool
+    return vol.Schema(spec)
+
+
+def get_image_type(image):
+    """Get the format of downloaded bytes that could be an image."""
+    fmt = imghdr.what(None, h=image)
+    if fmt is None:
+        # if imghdr can't figure it out, could be svg.
+        with contextlib.suppress(UnicodeDecodeError):
+            if image.decode("utf-8").startswith("<svg"):
+                return "svg+xml"
+    return fmt
+
+
+async def async_test_still(hass, info) -> tuple[dict[str, str], str | None]:
```

Comment:
Missing type annotations for signature parameters. Please complete the type annotations for all the functions in this module.